pkg/compat/client.go

```diff
@@ -109,31 +108,59 @@ func (c client) MinorVersion() int {
 
 //
 // supportedVersion will determine correct version of the object provided, based on cluster version
-func (c client) supportedVersion(obj runtime.Object) runtime.Object {
+func (c client) supportedVersion(obj k8sclient.Object) k8sclient.Object {
 	if c.Minor < 16 {
 		switch obj.(type) {
 
 		// Deployment
 		case *appsv1.Deployment:
 			return &appsv1beta1.Deployment{}
-		case *appsv1.DeploymentList:
```

Comment:
Why did re move the list types here?

Reply:
@sseago I would like to know this as well.

Reply:
Ah, I see it's implemented in 
```
supportedVersionList(obj k8sclient.ObjectList) k8sclient.ObjectList
```

Reply:
Why the change?

Reply:
@djwhatle the old API just used `runtime.Object` for everything. WIth the new API, it's using `client.Object` and `client.ObjectList`, so we're dealing with single objects and lists under different API methods. That required splitting this out in the same way. Note the return type differs between the two methods split out as well.